pkg/operator/ceph/cluster/cluster.go

```diff
@@ -124,6 +128,11 @@ func (c *cluster) detectCephMajorVersion(image string, timeout time.Duration) (s
 
 func (c *cluster) createInstance(rookImage string) error {
 
+	if c.orchestrationRunning == true {
```

Comment:
A couple of suggestions for the `createInstance()` method:
- Since multiple goroutines will update these bools, we'll need to use a lock to get and set them safely. 
- It would be good to avoid recursion if an orchestration is pending. How about a `while` loop based on `orchestrationPending`?

Reply:
I might suggest adding a `RWMutex` (or `Mutex`) in the `cluster` struct and make methods for `orchestrationIsRunning` and `orchestrationIsPending`, and also add setter methods like `setOrchestrationRunning(yesNo bool)` / `setOrchestrationPending(yesNo bool)`. Inside these methods, you can do the `Lock() ; defer Unlock()`.

Reply:
As a follow-up note: we will have to be careful to make sure the process of checking if orchestration is running, then locking the orchestration once an orchestration process begins needs to be atomic. Otherwise, two processes may read that `orchestrationRunning == false` before one of them can set `orchestrationRunning = true`.

I might suggest adding a `Mutex` in the `cluster` struct and make methods for `orchestrationIsRunning` and `orchestrationIsPending`, and also add setter methods like `setOrchestrationRunning(yesNo bool)` / `setOrchestrationPending(yesNo bool)`. Inside these methods, you can do the `Lock() ; defer Unlock()`.

Reply:
IMO doing that in a loop is a neater as on the one hand no thread gets locked and on the other hand we do not call createInstance needlessly often as we call it not as often as it got called but rather only as often as needed.